back/engines/commercial/smart_groups/app/lib/smart_groups/rules/participated_in_topic.rb

```diff
@@ -54,31 +79,31 @@ def filter users_scope
 
       case predicate
       when 'in'
-        participants = participants_service.topics_participants([Topic.find(value)])
+        participants = participants_service.topics_participants(Topic.where(id: value).to_a)
```

Comment:
Using `Topic.where...to_a`, not `[Topic.where...]` to return an array of Topics, not a collection of ActiveRecord::Relations (which will not play well with `def topics_participants` in `back/app/services/participants_service.rb`

Reply:
The problem with `.to_a` is that this will force the query `Topic.where(...)`to execute separately. This is not acceptable in terms of efficiency (smart groups are one of the main reasons some platforms are extremely slow today), the `filter` method should not run any query but should instead return `ActiveRecord::Relations`, which represents one big SQL query to get the users of that smart group.

I think instead we should fix `topics_participants` so it can deal with a collection of topics.

Reply:
Sure. The multi-value tests will pass if we use `ideas = Idea.with_some_topics(topics.map(&:ids))` in `topics_participants`, but how to test for a collection? It will be in the form `[#<ActiveRecord::Relation [Topics…]]` if we use `[Topic.where(id: value)]` in the predicate cases.

Reply:
I guess it would make more sense if `.with_some_topics` would work with a collection of topics rather than IDs. Indeed this is a big issue, it means that we already have this issue today. It's probably not going to be very easy fixing it, but I think we should and by doing it, some of our current platforms might get somewhat faster.